src/components/MessageCard/components/MessageCard.Survey.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  ConfirmationMessageUI,
+  FeedbackFormUI,
+  SpinnerContainerUI,
+  SurveyUI,
+} from './MessageCard.Survey.css'
+import Input from '../../Input'
+import Button from '../../Button'
+import Spinner from '../../Spinner'
+import Icon from '../../Icon'
+
+export const MessageCardSurvey = ({
+  children,
+  withFeedbackForm = false,
+  forceFeedbackForm = false,
+  feedbackFormText = '',
+  onSubmit = () => {},
+  showSpinner = false,
+  showConfirmationMessage = false,
+}) => {
+  const [feedback, setFeedback] = React.useState('')
+  const [selected, setSelected] = React.useState(null)
+  const [showFeedbackForm, setShowFeedbackForm] = React.useState(false)
+  const shouldShowFeedbackForm = showFeedbackForm || forceFeedbackForm
+
+  function handleSelection(id) {
+    setSelected(id)
+
+    if (withFeedbackForm) {
+      setShowFeedbackForm(true)
+      return
+    }
+
+    onSubmit({
+      selected: id,
+    })
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault()
+    onSubmit({
+      selected,
+      feedback,
+    })
+  }
+
+  function renderChildren() {
+    return React.Children.map(children, child => {
+      return React.cloneElement(child, {
+        selected,
+        onSelection: handleSelection,
+      })
+    })
+  }
+
+  if (showConfirmationMessage) {
+    return (
+      <SurveyUI data-cy="beacon-message-cta-survey">
+        <ConfirmationMessageUI>
+          <Icon name="tick-small" inline size={24} />
+          Thanks for the feedback
+        </ConfirmationMessageUI>
+      </SurveyUI>
+    )
+  }
+
+  return (
+    <SurveyUI data-cy="beacon-message-cta-survey">
+      {renderChildren()}
+
+      {shouldShowFeedbackForm && (
+        <FeedbackFormUI onSubmit={handleSubmit}>
+          <label htmlFor="survey-comment">{feedbackFormText}</label>
+          <Input
+            id="survey-comment"
+            name="survey-comment"
+            withCharValidator={true}
+            charValidatorLimit={120}
+            multiline={3}
+            value={feedback}
+            onChange={value => setFeedback(value)}
+          />
+          <Button submit theme="blue">
+            Send
+          </Button>
+        </FeedbackFormUI>
+      )}
+
+      {showSpinner && (
```

Comment:
Should the spinner replace feedback form? If I understand it correctly it should be shown when someone clicked `Send`, right?

Reply:
The spinner doesn't replace the form, it's an overlay on top of it (at least for the POC, but it's subject to change in the future).

<img width="338" alt="CleanShot 2022-02-01 at 10 36 08@2x" src="https://user-images.githubusercontent.com/44473/152029911-46d487e1-74af-41db-be74-43d37828e960.png">